lib/oxrdf/src/blank_node.rs

```diff
@@ -121,6 +126,49 @@ impl Default for BlankNode {
     }
 }
 
+#[cfg(feature = "serde")]
+impl Serialize for BlankNode {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut state = serializer.serialize_struct("BlankNode", 1)?;
+        state.serialize_field("value", &self.as_str())?;
+        state.end()
+    }
+}
+
+#[cfg(feature = "serde")]
+struct BlankNodeVisitor;
+
+#[cfg(feature = "serde")]
+impl<'de> Visitor<'de> for BlankNodeVisitor {
+    type Value = BlankNode;
+    fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        formatter.write_str("struct BlankNode")
+    }
+    fn visit_map<V>(self, mut map: V) -> Result<BlankNode, V::Error>
+    where
+        V: MapAccess<'de>,
+    {
+        let key = map.next_key::<String>()?;
+        if key != Some("value".to_string()) {
+            if let Some(val) = key {
+                return Err(de::Error::unknown_field(&val, &["value"]));
+            }
+            return Err(de::Error::missing_field("value"));
+        }
+        Ok(BlankNode::new(map.next_value::<String>()?).map_err(de::Error::custom)?)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for BlankNode {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_struct("BlankNode", &["value"], BlankNodeVisitor)
```

Comment:
Can we make it much simpler with something like (not sure about the reference)?
```rust
#[derive(Deserialize)]
struct BNode<'a> {
    value: &'a str
}
let bnode = Bnode::deserialize(deserializer)?;
Self::new(bnode.value).map_err(de::Error::custom)
```
(same for IRIs)

Reply:
> the implementation will no longer error if unexpected properties are present in the object being deserialised

You should be able to avoid this by adding the [`deny_unknown_fields`](https://serde.rs/container-attrs.html#deny_unknown_fields) to the intermediate struct.

Reply:
Fine w. me - lets leave as is.